Cilsil/Cil/Parsers/ExceptionParser.cs

```diff
@@ -0,0 +1,293 @@
+                PRUNE(n$49, true); [line 55];
+                *&amp;CatchVar65:void=n$48 [line 55];*/
+
+                /*node3 preds:1 succs:5 exn: Conditional exception handler -- Connects to outter/finally block code directly
+                n$49=_fun___instanceof(n$48:java.io.IOException*,sizeof(t=java.io.IOException):void) [line 55];
+                PRUNE(!n$49, false); [line 55];*/ 
+
+                // n${i+1}=_fun___instanceof(n${i}:{Exception_Type}*,sizeof(t={Exception_Type}:void)
+                (var exceptionAllocationCall, var objectVariable) = 
+                        CreateExceptionAllocationCall(state, Typ.FromTypeReference(exceptionType));   
+
+                var trueNodeCondition = objectVariable;
+                var falseNodeCondition = new UnopExpression(
+                    UnopExpression.UnopKind.LNot, trueNodeCondition, null);
+                (var exceptionTrueNode, var exceptionFalseNode) =
+                    CreateExceptionBranchConditionNodes(state,
+                                                        trueNodeCondition,
+                                                        falseNodeCondition);
+
+                exceptionTrueNode.Instructions.Insert(0, exceptionAllocationCall);
+                exceptionFalseNode.Instructions.Insert(0, exceptionAllocationCall);
+                
+                // Create catch exception variable expression
+                catchVariable = new LvarExpression(
+                                new LocalVariable(Identifier.CatchIdentifier,
+                                                    state.Method));
+                var catchVarIdentifier = state.GetIdentifier(Identifier.IdentKind.Normal);
+                var catchVarStore = new Store(catchVariable,
+                                              new VarExpression(returnVariable),
+                                              expressionType,
+                                              state.CurrentLocation);
+                exceptionTrueNode.Instructions.Add(catchVarStore);
+
+                RegisterNode(state, exceptionTrueNode);
+                RegisterNode(state, exceptionFalseNode);
+
+                Instruction targetFalse = null;
+                
+                foreach(var single_instruction in state.Method.Body.Instructions)
+                {
+                    if (single_instruction.Offset >= endBlockOffset + 1)
+                    {
+                        targetFalse = single_instruction;
+                        break;
+                    }
+                }
+                state.PushExpr(new VarExpression(returnVariable), expressionType);
+
+                state.PushInstruction(targetFalse, exceptionFalseNode);
+
+                state.PushInstruction(instruction, exceptionTrueNode);
+                instruction = state.PopInstruction();
+            }
+            else
+            {
+                if (instruction.Next.OpCode.Code == Code.Brfalse_S)
```

Comment:
Comment that this is specifically to handle the "using" case,  which starts with a Brfalse_s

Reply:
Done